src/stan/mcmc/chainset.hpp

```diff
@@ -0,0 +1,510 @@
+    return (draws.array() - draws.mean()).square().sum() / (draws.size() - 1);
+  }
+
+  /**
+   * Compute sample variance for specified parameter across all chains.
+   * 1 / (N - 1) * sum((theta_n - mean(theta))^2)
+   *
+   * @param name parameter name
+   * @return sample variance
+   */
+  double variance(const std::string& name) const {
+    return variance(index(name));
+  }
+
+  /**
+   * Compute standard deviation for specified parameter across all chains.
+   *
+   * @param index parameter index
+   * @return sample sd
+   */
+  double sd(const int index) const { return std::sqrt(variance(index)); }
+
+  /**
+   * Compute standard deviation for specified parameter across all chains.
+   *
+   * @param name parameter name
+   * @return sample sd
+   */
+  double sd(const std::string& name) const { return sd(index(name)); }
+
+  /**
+   * Compute median value of specified parameter across all chains.
+   *
+   * @param index parameter index
+   * @return median
+   */
+  double median(const int index) const { return (quantile(index, 0.5)); }
+
+  /**
+   * Compute median value of specified parameter across all chains.
+   *
+   * @param name parameter name
+   * @return median
+   */
+  double median(const std::string& name) const { return median(index(name)); }
+
+  /**
+   * Compute maximum absolute deviation (mad) for specified parameter.
+   *
+   * Follows R implementation:  constant * median(abs(x - center))
+   * where the value of center is median(x) and the constant is 1.4826,
+   * a scale factor for asymptotically normal consistency: `1/qnorm(3/4)`.
+   * (R stats version 3.6.2)
+   *
+   * @param index parameter index
+   * @return sample mad
+   */
+  double max_abs_deviation(const int index) const {
+    Eigen::MatrixXd draws = samples(index);
+    auto center = median(index);
+    Eigen::MatrixXd abs_dev = (draws.array() - center).abs();
+    Eigen::Map<Eigen::VectorXd> map(abs_dev.data(), abs_dev.size());
+    return 1.4826 * stan::math::quantile(map, 0.5);
+  }
+
+  /**
+   * Compute maximum absolute deviation (mad) for specified parameter.
+   *
+   * Follows R implementation:  constant * median(abs(x - center))
+   * where the value of center is median(x) and the constant is 1.4826,
+   * a scale factor for asymptotically normal consistency: `1/qnorm(3/4)`.
+   * (R stats version 3.6.2)
+   *
+   * @param name parameter name
+   * @return sample mad
+   */
+  double max_abs_deviation(const std::string& name) const {
+    return max_abs_deviation(index(name));
+  }
+
+  /**
+   * Compute the quantile value of the specified parameter
+   * at the specified probability.
+   *
+   * Throws exception if specified probability is not between 0 and 1.
+   *
+   * @param index parameter index
+   * @param prob probability
+   * @return parameter value at quantile
+   */
+  double quantile(const int index, const double prob) const {
+    // Ensure the probability is within [0, 1]
+    if (prob <= 0.0 || prob >= 1.0) {
```

Comment:
stan::math::quantile already checks its bounds, so I don't know if this is necessary. At any rate, it actually allows 0 and 1, so we probably should too.